File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """

        new_dict = {}
        complement = 0
        
        for i in range(len(nums)):
            complement = target - nums[i]
            if (complement in new_dict):
                return [i, new_dict[complement]]
            new_dict[nums[i]] = i 
        
        return -1
        

        

 

                

        

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> myHashMap = new HashMap<>();

        for(int i = 0; i< nums.length; i++){
            int num = nums[i];
            int complement = target - num;

            if(myHashMap.containsKey(complement)){
                return new int[]{myHashMap.get(complement), i};
            }
            myHashMap.put(num, i);
        }
        return new int[]{};
        
    }
}